internal/controller/backend_security_policy.go

```diff
@@ -43,7 +46,31 @@ func (b backendSecurityPolicyController) Reconcile(ctx context.Context, req ctrl
 		return ctrl.Result{}, err
 	}
 
+	if getBackendSecurityPolicyAuthOIDC(backendSecurityPolicy.Spec) != nil {
+		var requeue time.Duration
+		requeue = time.Minute
+		region := backendSecurityPolicy.Spec.AWSCredentials.Region
+		rotator, err := backendauthrotators.NewAWSOIDCRotator(b.client, b.kube, b.logger, backendSecurityPolicy.Namespace, backendSecurityPolicy.Name, preRotationWindow, region)
+		if err != nil {
+			b.logger.Error(err, "failed to create AWS OIDC rotator")
+		} else if !rotator.IsExpired() {
+			requeue = time.Until(*rotator.GetPreRotationTime())
+			if requeue.Seconds() == 0 {
+				requeue = time.Minute
+			}
+		}
+		res = ctrl.Result{RequeueAfter: requeue, Requeue: true}
```

Comment:
looks like the comment on `RequeueAfter` says there's no need to set Requeue=true
```suggestion
		res = ctrl.Result{RequeueAfter: requeue}
```

> 	// RequeueAfter if greater than 0, tells the Controller to requeue the reconcile key after the Duration.
	// Implies that Requeue is true, there is no need to set Requeue to true at the same time as RequeueAfter.
	RequeueAfter time.Duration

Reply:
On a side note, we may have a queuing / race condition problem. Reconciles will always requeue a BSP event if it specifies OIDC. Each time a change happens in that BSP (ie region changes), we will add requeue another event to be reconciled.  There will be N different versions of the same BSP if N changes were made.

Still looking into this but ideally the controller can check what is new vs requeued and removed stale events.

Reply:
please leave it as a TODO, you do that in a follow up...